cdap-proto/src/test/java/co/cask/cdap/proto/ProtoTriggerCodecTest.java

```diff
@@ -61,6 +63,16 @@ public void testTriggerCodec() {
     json = GSON.toJson(trigger, Trigger.class);
     trigger1 = GSON.fromJson(json, Trigger.class);
     Assert.assertEquals(trigger, trigger1);
+
+    trigger = new ProtoTrigger.ProgramStatusTrigger(new ProgramId("test", "myapp",
+                                                    ProgramType.FLOW, "myprog"),
+                                                    ImmutableSet.of(co.cask.cdap.api.ProgramStatus.FAILED));
+    json = GSON.toJson(trigger);
```

Comment:
you can also create a helper method to take a trigger and complete all the following comparisons